hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieCDCFileIndex.scala

```diff
@@ -42,29 +42,34 @@ class HoodieCDCFileIndex (override val spark: SparkSession,
   extends HoodieIncrementalFileIndex(
     spark, metaClient, schemaSpec, options, fileStatusCache, includeLogFiles, shouldEmbedFileSlices
   ) with FileIndex {
+  private val emptyPartitionPath: String = "empty_partition_path";
   val cdcRelation: CDCRelation = CDCRelation.getCDCRelation(spark.sqlContext, metaClient, options)
   val cdcExtractor: HoodieCDCExtractor = cdcRelation.cdcExtractor
 
   override def listFiles(partitionFilters: Seq[Expression], dataFilters: Seq[Expression]): Seq[PartitionDirectory] = {
-    val partitionToFileGroups = cdcExtractor.extractCDCFileSplits().asScala.groupBy(_._1.getPartitionPath).toSeq
-    partitionToFileGroups.map {
-      case (partitionPath, fileGroups) =>
-        val fileGroupIds: List[FileStatus] = fileGroups.map { fileGroup => {
-          // We create a fake FileStatus to wrap the information of HoodieFileGroupId, which are used
-          // later to retrieve the corresponding CDC file group splits.
-          val fileGroupId: HoodieFileGroupId = fileGroup._1
-          new FileStatus(0, true, 0, 0, 0,
-            0, null, "", "", null,
-            new Path(fileGroupId.getPartitionPath, fileGroupId.getFileId))
-        }}.toList
-        val partitionValues: InternalRow = new GenericInternalRow(doParsePartitionColumnValues(
-          metaClient.getTableConfig.getPartitionFields.get(), partitionPath).asInstanceOf[Array[Any]])
+    cdcExtractor.extractCDCFileSplits().asScala.map {
+      case (fileGroupId, fileSplits) =>
+        val partitionPath = if (fileGroupId.getPartitionPath.isEmpty) emptyPartitionPath else fileGroupId.getPartitionPath
```

Comment:
using empty String instead for non-partitioned table?

Reply:
Here we cannot use empty string since Line 63 requires the partition_path to be not empty.